### ISSUE TITLE
The Amazing Spider-Man - GPU crash

### ISSUE THREAD

**Issue by Electry** [Electry/VitaGrafix#122]
Running the game at 960x544 casues GPU crash in some areas.
I suspect it's due to VRAM limitations (again).

**Comment by pspman2008** [comment]
o my god.you are so incredible!!!!

**Comment by chrmichael** [comment]
Well - the patch does not patch vertical res. I suspect that this is the reason. For horisontal res, anything goes, as long as OT is a multiple of 32, possibly reflecting that the game runs in 16-bit mode..
Internal buffer is probably aligned to 64  bytes.
The GPU crash comes when pressing select (the upgrade abiæityty menu)

**Comment by Peoples00** [comment]
can you fix that?

**Comment by justalex95** [comment]
hey try this, set the resolution to: 960x544,1280x720, if that doesnt wrk try treating it like it has dynamic resolution and try 3 resolutions under native.

running 960x544,1280x720, mennus open and everything works fine, incl, city swinging

**Comment by eddyfire** [comment]
you meant typing the whole "960x544,1280x720" on IB? or just 960x544? 
because tried 960x544 theres no diff and gpu crashes

**Comment by justalex95** [comment]
type '960x544,1280x720' its the format for dynamic resolutions in vitagrafix, I thought i'd see if spiderman could do dynamic resolutions, it probably doesnt do anything dynamically but now I have it running with '960x544,1280x720' as its resolution. 

dont get me wrong its not running at 1280x720, because that breaks the graphics, its running at native res but without the menus crashing. the res should show as 960x544 >> 1280x720

**Comment by eddyfire** [comment]
yup it works now! no crash so far. thanks @justalex95!

**Comment by justalex95** [comment]
glad it works, I'm 100% sure its running at native res. I'm using 960x544,1920x1080 just now and it feels a little less jerky with the loading, it sounds wacky but around the city its definitely smoother

**Comment by chrmichael** [comment]
hmm.. the second resolution does absolutely nothing. I think it is ignored by the patch.

704x400 work as IB.

960x544 works for the first level, at least. Also in menu and swinging in manhattan (2nd level).

I never tried 960x544, because Electry wrote that it produces crashes in some areas.

I was interested in running this at lower resolutions like 640x368 or 576x324 to gain fps, but these crash in the menu.

**Comment by justalex95** [comment]
Yeah, the second resolution does nothing but if you put 960x544 on its own, it crashes when you press select to open the menu, for some reason with more than one resolution it doesnt crash the select menu, making it playable on max resolution

**Comment by Electry** [comment]
Grab [VitaGrafix.zip](https://github.com/Electry/VitaGrafixPatchlist/files/3800902/VitaGrafix.zip) and try this:

```
# The Amazing Spider-Man
[PCSE00333, eboot.bin] # [US 1.00]
[PCSB00428, eboot.bin, 0xE5988D4F] # [EU 1.00]
@IB
# lpddr force 221MB
0:0x6C9A2 t2_mov(0, 8, 0xDD00000)
# RT
0:0x1769AA t2_mov(1, 1, ib_w)
0:0x1769AE t2_mov(1, 2, ib_h)
0:0x176ADE t2_mov(1, 6, ib_w)
0:0x176B22 t2_mov(1, 6, ib_h)
0:0x16CD7E t2_mov(1, 1, ib_w)
0:0x16CD82 t2_mov(1, 0, ib_h)
0:0x22F1C8 t2_mov(1, 2, ib_w)
0:0x22F1CE t2_mov(1, 3, ib_h)
# Menu
0:0x655D52 t2_mov(1, 0, ib_w)
0:0x655D5A t2_mov(1, 1, ib_h)
0:0x655D78 t2_mov(1, 2, ib_w)
0:0x655D7E t2_mov(1, 3, ib_h)
# Tablet
0:0x16D008 t1_movt(1, float(ib_w) >> 16)
0:0x16D014 t1_movt(1, float(ib_h) >> 16)
0:0x19240A t1_movt(0, float(ib_w) >> 16)
0:0x19241E t1_movt(0, float(ib_h) >> 16)
@FPS
>sceDisplaySetFrameBuf_withWait()
```

You can either replace the old patch in patchlist.txt or create **ux0:data/VitaGrafix/patch/PCSB00428.txt** file and copy it there temporarily.

**Comment by justalex95** [comment]
@electry, in config.txt do you need to specify 60 fps or does the code enforce this automatically?

Going to try it out later

**Comment by justalex95** [comment]
tried it out, the game crashes when opening the menu still

**Comment by justalex95** [comment]
Wow, just wow, The EU version works like a charm and it might be my imagination but did you do something with the low resolution in the distance of each scene? 

The game runs a ton smoother, looks great and I dont know if its because its running smoother but the graphics look slightly more textured, spiderman just looks sharp as hell and the boss that charges at you in the car park is completeable - It wouldnt let me hit him the third time to beat him, it kept bugging out before.

well done man, this is absolutely superb.

I did have a crash and game exit (not a complete system freeze) when I left my flat for the first time but I decided not to skip the scene the second time and it was great after that.

Thats as far as I've done so far but if this is the new version of vitagrafix I have to take my hat off @Electry 

if you have a patreon I'll be happy to add to it, just brilliant.

its not really worth mentioning but the US version doesnt work, only the EU version works currently. The US version still has the menu bug
